src/bitmap.c

```diff
@@ -1693,7 +1693,8 @@ gdip_pixel_stream_copy_optimized (StreamingState *dst_state, StreamingState *src
 
 	/* if the target is in 32bpp format with only RGB but no alpha then we should forcefully set all alpha bits to 0xff -> optimize this later! */
 	if (dst_state->pixels_per_byte == -4) {
-		if (dst_state->data->pixel_format == PixelFormat32bppRGB) return FALSE;
+		if (dst_state->data->pixel_format == PixelFormat32bppRGB && src_state->data->pixel_format != PixelFormat32bppRGB)
```

Comment:
`test_bitmapSetPixel32bppRGB`: `GdipBitmapLockBits` forced setting alpha bits to 0xFF, which did not match GDI+ behaviour. Fixed by adding `dst_state->data->pixel_format == PixelFormat32bppRGB && src_state->data->pixel_format != PixelFormat32bppRGB`